### PR TITLE
CRC64 not optional in UploadRangeFromURL

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azfile/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 ### Breaking Changes
 
+* Changed type of SourceContentCRC64 header from uint64 to a generic interface implementation, so that it is considered
```
Comment (vibhansa-msft) [inline sdk/storage/azfile/CHANGELOG.md:9]: I think we mention the breaking changes seperatly. Just check if this is the correct format or not.

File: sdk/storage/azfile/file/client_test.go [hunk 2]
```diff
@@ -4256,8 +4255,8 @@ func (f *FileUnrecordedTestsSuite) TestFileUploadRangeFromURLNow() {
 	_require.NotNil(cResp.FileLastWriteTime)
 
 	uResp, err := destFClient.UploadRangeFromURL(context.Background(), srcFileSAS, 0, 0, int64(contentSize), &file.UploadRangeFromURLOptions{
-		SourceContentCRC64: contentCRC64,
-		LastWrittenMode:    to.Ptr(file.LastWrittenModeNow),
+		SourceContentValidation: file.SourceContentValidationTypeCRC64(contentCRC64),
```
Comment (souravgupta-msft) [inline sdk/storage/azfile/file/client_test.go:4258]: Let's also add a test where `SourceContentValidation` is nil in the `UploadRangeFromURLOptions`.

Comment (tanyasethi-msft) [reply sdk/storage/azfile/file/client_test.go:4324]: Added

**Review by souravgupta-msft — APPROVED** [review]
